src/model/Surface.cpp

```diff
@@ -1440,16 +1440,101 @@ namespace model {
+      std::vector<Point3d> parentVertices = parentToXY.inverse() * this->vertices();
+      // Make sure the parent surface is oriented clockwise
+      auto norm = openstudio::getOutwardNormal(parentVertices);
+      if (norm && norm->z() > 0) {
+        std::reverse(parentVertices.begin(), parentVertices.end());
+      }
+
+      // Get the flattened sub surface vertices for windows
       for (const SubSurface& subSurface : this->subSurfaces()) {
         if (istringEqual(subSurface.subSurfaceType(), "FixedWindow") || istringEqual(subSurface.subSurfaceType(), "OperableWindow")) {
-          windowArea += subSurface.multiplier() * subSurface.netArea();
+          auto opening = parentToXY.inverse() * subSurface.roughOpeningVertices();
+          auto norm = openstudio::getOutwardNormal(opening);
+          if (norm && norm->z() > 0) {
+            std::reverse(opening.begin(), opening.end());
+          }
+          roughOpenings[subSurface] = opening;
         }
       }
 
-      double wwr = windowArea / grossArea;
+      // Check each rough opening against the parent surface, if any vertices are outside
+      // the parent surfave then revert back to the original vertices
+      for (const auto& opening : roughOpenings) {
+        const SubSurface& subSurface = opening.first;
+        std::vector<Point3d> vertices = opening.second;
 
-      return wwr;
+        if (!openstudio::polygonInPolygon(vertices, parentVertices, tol)) {
+          // One or more vertices not within the parent surface so replace with the original vertices
+          auto vertices = parentToXY * subSurface.vertices();
+          auto norm = openstudio::getOutwardNormal(vertices);
+          if (norm && norm->z() > 0) {
+            std::reverse(vertices.begin(), vertices.end());
+          }
+          roughOpenings[subSurface] = vertices;
+          break;
+        }
+      }
+
+      // Check subsurfaces for overlaps
+      // (if I knew how I'd order themn left to right in x then we could test adjacent ones only)
+      for (const auto& opening1 : roughOpenings) {
+        for (const auto& opening2 : roughOpenings) {
+          if (opening1 == opening2) continue;
+
+          auto result = openstudio::intersect(opening1.second, opening2.second, tol);
+          if (result) {
+            //There's an overlap so swap out the overlapped vertices for the original
+            auto vertices = parentToXY * opening1.first.vertices();
+            auto norm = openstudio::getOutwardNormal(vertices);
+            if (norm && norm->z() > 0) {
+              std::reverse(vertices.begin(), vertices.end());
+            }
+            roughOpenings[opening1.first] = vertices;
+          }
+        }
+      }
+
+      // Accumulate the areas
+      double area = 0;
+      for (const auto& roughOpening : roughOpenings) {
+        area += openstudio::getArea(roughOpening.second).value() * roughOpening.first.multiplier();
+      }
```

Comment:
Maybe it doesn't really improve the lisibility, but std::accumulate (`#include <numeric>`) is meant for that.

```suggestion
      // Accumulate the areas
      double area = std::accumulate(roughOpenings.cbegin(), roughOpenings.cend(), 0.0, [](double sum, auto& roughOpening) {
        return sum + openstudio::getArea(roughOpening.second).value() * roughOpening.first.multiplier();
      });
```